src/features/ImeiProgramming/index.tsx

```diff
@@ -0,0 +1,433 @@
+    DeviceInfo,
+    NrfutilDeviceLib,
+} from '@nordicsemiconductor/pc-nrfconnect-shared/nrfutil/device';
+import { clipboard } from 'electron';
+import path from 'path';
+
+import { readIMEI, writeIMEI } from '../nrfutil91';
+import fetchIMEI from './fetchIMEI';
+
+const PasswordComponent = ({
+    onChanging,
+    value,
+}: {
+    onChanging: (value: string) => void;
+    value: string;
+}) => {
+    const [show, setShow] = useState(false);
+    return (
+        <div className="tw-flex tw-flex-row">
+            <input
+                className="tw-w-28 tw-border tw-border-r-0 tw-border-gray-400 tw-px-2 tw-py-1 tw-leading-none"
+                type={show ? 'text' : 'password'}
+                value={value}
+                onChange={event => {
+                    onChanging(event.target.value);
+                }}
+            />
+            <button
+                onClick={() => setShow(!show)}
+                className="tw-border tw-border-l-0 tw-border-gray-400"
+                type="button"
+            >
+                <span
+                    className={`mdi tw-leading-none ${
+                        !show ? ' mdi-eye-off' : ' mdi-eye'
+                    }`}
+                />
+            </button>
+        </div>
+    );
+};
+
+const isSupportedDevice = (deviceInfo?: DeviceInfo) =>
+    deviceInfo?.jlink?.protectionStatus === 'NRFDL_PROTECTION_STATUS_NONE' &&
+    deviceInfo?.jlink?.deviceVersion?.toLocaleUpperCase().match(/NRF91\d1/);
+const isMaybeSupportedDevice = (deviceInfo?: DeviceInfo) =>
+    deviceInfo?.jlink?.protectionStatus === 'NRFDL_PROTECTION_STATUS_NONE'
+        ? deviceInfo?.jlink?.deviceVersion
+              ?.toLocaleUpperCase()
+              .match(/NRF91\d1/)
+        : deviceInfo?.jlink?.deviceFamily !== 'NRF91_FAMILY';
+
+const getStatus = async (device?: Device, deviceInfo?: DeviceInfo) => {
+    if (!device || !deviceInfo) return 'NONE';
+    if (deviceInfo.jlink?.protectionStatus !== 'NRFDL_PROTECTION_STATUS_NONE')
+        return 'PROTECTED';
+
+    if (!device.serialNumber || !isSupportedDevice(deviceInfo))
+        return 'UNSUPPORTED';
+
+    const defaultImeiNumber = 'FFFFFFFFFFFFFFF';
+    try {
+        const res = await readIMEI(device.serialNumber);
+        if (res !== defaultImeiNumber) {
+            return 'IMEI_SET';
+        }
+        return 'IMEI_NOT_SET';
+    } catch (e) {
+        return 'FIRMWARE';
+    }
+};
+
+const updateDeviceInfo = async (device: Device) => {
+    const deviceInfo = await NrfutilDeviceLib.deviceInfo(device);
+    setSelectedDeviceInfo(deviceInfo);
+    return deviceInfo;
+};
+
+const recover = async (device: Device) => {
+    await NrfutilDeviceLib.batch()
+        .recover('Application')
+        .reset('Application')
+        .run(device);
+    return getStatus(device, await updateDeviceInfo(device));
+};
+
+const CopyImeiNotice = ({ imei }: { imei: string }) => (
+    <>
+        Copy and store the IMEI as it has been consumed from the cloud.
```

Comment:
```suggestion
        Copy and store the IMEI. Numbers consumed by the cloud cannot be reused.
```
or
```suggestion
        The IMEI has been consumed in the cloud. Copy and store it.
```
I think what bothers me is not necessarily "consumed", but this weird "from". Why "from" and not "by" or "in"?